zio/csvio/reader.go

```diff
@@ -29,9 +33,12 @@ type Reader struct {
 //	StringsOnly bool
 //}
 
-func NewReader(zctx *zed.Context, r io.Reader) *Reader {
+func NewReader(zctx *zed.Context, r io.Reader, opts ReaderOpts) *Reader {
 	preprocess := newPreprocess(r)
 	reader := csv.NewReader(preprocess)
+	if opts.Delim != ',' {
```

Comment:
We want reasonable default behavior with the zero value for opts (so that `csvio.NewReader(zctx, r, csvio.ReaderOpts{})` returns a usable reader).
```suggestion
	if opts.Delim != 0 {
```